tests/js/realm-tests.js

```diff
@@ -970,8 +968,7 @@ module.exports = {
         const realm2 = new Realm({schema: [schemas.StringOnly], shouldCompactOnLaunch: shouldCompact});
         TestCase.assertTrue(wasCalled);
         TestCase.assertEqual(realm2.objects('StringOnlyObject').length, count + 2);
-        var afterSize = fs.statSync(realm2.path).size;
-        TestCase.assertTrue(beforeSize >= afterSize);
+        // we don't check if the file is smaller as we assume that Object Store does it
```

Comment:
Here we only check that the callback given to Object Store is actually called.IMHO, we don't have to test if Object Store is doing the right thing - that's should be done by Object Store's tests.